### PR TITLE
[Master] Separate main from Master

### PR DESCRIPTION
**Pull request by Yuzhen11**

### CODE REVIEW

File: master/master.cpp [hunk 1]
```diff
@@ -77,18 +76,3 @@ void Master::handle_message(uint32_t message, const std::string& id) {
 
 }  // namespace husky
```
Comment (kygx-legend) [inline master/master.cpp]: Don't leave an empty line in the end of a file.

**Comment by Yuzhen11** [comment]
OK now. Ready to be merged. 

If we want static global variables to be initialized before the main function, we shouldn't archive the translation units containing the static variable. Otherwise, the global variables will not be initialized. 

So later we should be careful about the trick to use global variables to register lambda.  The point is that we can compile them into object files but not archive.

File: master/CMakeLists.txt [hunk 2]
```diff
@@ -25,11 +25,14 @@ if(MONGOCLIENT_FOUND)
...
-set(husky husky-lib husky-io husky-core husky-base)
+add_library(husky-master-objs OBJECT master.cpp ${master_plugins})
+husky_default_properties(husky-master-objs)
+
+set(husky-master-dep husky-io husky-core husky-base)
 
 # Master
-add_executable(Master master.cpp ${master_plugins})
-target_link_libraries(Master ${husky})
+add_executable(Master master_main.cpp $<TARGET_OBJECTS:husky-master-objs>)
```
Comment (kygx-legend) [inline master/CMakeLists.txt]: I think it also need to call `husky_default_properties` for the executable target.

Comment (kygx-legend) [inline master/CMakeLists.txt]: A little confused about the variable after .cpp. What does that do?

Comment (Yuzhen11) [inline master/CMakeLists.txt]: The `husky_default_properties` is in the end of the file, please check.

The variable after .cpp are the objects that contain global variables which need to be initialized.

Comment (kygx-legend) [inline master/CMakeLists.txt]: OIC.

**Pull request merged**